core/src/main/scala/scalaz/MonadError.scala

```diff
@@ -11,6 +11,9 @@ trait MonadError[F[_], S] extends Monad[F] { self =>
   def raiseError[A](e: S): F[A]
   def handleError[A](fa: F[A])(f: S => F[A]): F[A]
 
+  def emap[A](fa: F[A])(f: A => S \/ A): F[A] =
+    bind(fa)(a => f(a).fold(raiseError(_), pure(_)))
```

Comment:
why `emap`? the name makes me think "error map" which would be `leftMap` if this is also a `Bifunctor`, but here it's more along the lines of "map with possible error represented by `\/`"

Reply:
that's the convention some library authors have been using... if anybody has a better name to suggest, I'm all ears. dmap, mapd, etc...